src/control/system/membership.go

```diff
@@ -150,6 +160,10 @@ func (m *Membership) Join(req *JoinRequest) (resp *JoinResponse, err error) {
 		return nil, err
 	}
 
+	if err := m.checkReqFaultDomain(req); err != nil {
```

Comment:
don't we need to do the same check as above for count? also why don't we just do the check once before checking err == nil? I'm probably missing something obvious.

Reply:
I started off that way, but then I found a special case where a member rejoins, but it's the only member. In that case the depth of the fault domain doesn't matter -- there are no other branches to compare it to, so it should be allowed to override the old fault domain.